likecoin/admin/matters.php

```diff
@@ -352,6 +352,66 @@ function likecoin_post_attachment_to_matters( $attachment_id ) {
 	return $matters_attachment_id;
 }
 
+/**
+ * Upload a file as draft image to matters
+ *
+ * @param int|     $image_url image url to be uploaded to matters.
+ * @param object | $image_infos all images' information of the post.
+ */
+function likecoin_post_url_image_to_matters( $image_url, $image_infos ) {
+	global $post;
+	$post_id              = $post->ID;
+	$file_path            = $image_url;
+	$file_path_components = preg_split( '#[.]#', $file_path );
+	$file_mime_type       = 'image/' . end( $file_path_components );
+	$filename             = basename( $file_path );
+	$parent_post          = $post_id;
+	$matters_info         = get_post_meta( $parent_post, LC_MATTERS_INFO, true );
+	if ( ! $matters_info ) {
+		$matters_info = array(
+			'type' => 'post',
+		);
+	}
+	if ( isset( $matters_info['published'] ) && $matters_info['published'] ) {
+		return;
+	}
+	$matters_draft_id = isset( $matters_info['draft_id'] ) ? $matters_info['draft_id'] : null;
+	if ( ! $matters_draft_id ) {
+		if ( ! likecoin_get_admin_errors() ) {
+			likecoin_handle_matters_api_error( 'Cannot save draft before publishing' );
+		}
+		return;
+	}
+	$attachment_type = 'image';
+	$api             = LikeCoin_Matters_API::get_instance();
+	$res             = $api->post_attachment(
+		array(
+			'path'      => $file_path,
+			'filename'  => $filename,
+			'mime_type' => $file_mime_type,
+			'type'      => $attachment_type,
+		),
+		$matters_draft_id
+	);
+	if ( isset( $res['error'] ) ) {
+		likecoin_handle_matters_api_error( $res['error'] );
+		return;
+	}
+	$matters_attachment_id = $res['id'];
+	if ( empty( $image_infos ) ) { // no picture has been uploaded to matters for this post.
+		$image_infos = array(); // initiate the array to avoid empty string can not push error.
+	}
+	$image_info                = (object) array(
+		'original_url'          => $image_url,
+		'matters_url'           => $res['path'],
+		'matters_attachment_id' => $res['id'],
+	);
+	$image_infos               = (array) $image_infos; // turn object into array so we can add properties.
+	$image_infos[ $image_url ] = $image_info;
```

Comment:
Does something like `$image_infos->{$image_url}` work, without the need to cast the info to array?

Reply:
Yes, it works. Thanks for the suggestion :)